internal/cmd/exodus.go

```diff
@@ -157,8 +157,20 @@ func exodusMain(ctx context.Context, cfg conf.Config, args args.Config) int {
 	for _, item := range items {
 		gwItem := gw.ItemInput{WebURI: webURI(item.SrcPath, args.Src, destTree)}
 
+		// When exodus-rsync is run with --links, publish items include link_to rather than object_key.
 		if item.LinkTo != "" {
-			gwItem.LinkTo = path.Join(destTree, strings.TrimLeft(item.LinkTo, "./"))
+			// First, attempt to resolve the symlink to another item in the source tree.
```

Comment:
I think this is barking up the wrong tree, it should be more like https://go.dev/play/p/_9GdImdYirI .

You probably sensed already that this is not quite right, still I'll add a couple of more points:

- we should not be resolving the symlinks client-side at all here.  If the link has enough levels of "../" to escape the transfer root, then it can end up successfully resolving to content which is entirely unrelated to the push.
- logically if you expect the "can't resolve symlink locally" path to work, then why would you need two code paths at all instead of using that one all the time?

Reply:
Thank you very much, I am still confused though. More questions to follow.

I actually tried something similar initially, but this solution fails for the existing rand1 and rand2 test cases (the link has enough levels of "../" to escape the transfer root).

```
  (item.LinkTo)             ../../../rand2
  (linkDestDirty)           /some/target/subdir/../../../rand2
  (actualResolvedLink)      /rand2
  (expectedLink)            /some/target/subdir/rand2
```

It is now obvious **why** /some/target/subdir/../../../rand2 resolves to /rand2 (I didn't count the number of "../" present in the dirty link, so the cause of the error didn't occur to me). It is still not obvious to me what to do about it, though. You say that we shouldn't be resolving the symlinks client-side at all here (though it seems like that is what you were doing in your example), what should we be doing then?

Reply:
> this solution fails for the existing rand1 and rand2 test cases

Which test exactly?  TestMainSyncDontFollowLinks?

The expectedItems in that test currently seem to be wrong.  In the bit you pasted, I don't know where expectedLink came from or why that would be expected.  actualResolvedLink seems to be the correct value and I'm not sure what you mean regarding what to do about it.

> You say that we shouldn't be resolving the symlinks client-side at all here

Sorry, what I meant is that we should not be asking the OS to follow them.

Reply:
You addressed both of my points of confusion, so I think that clears up all of my questions. Thanks again!